### PR TITLE
Update the interfaces about folly::future and fbthrift constructor

### PR DESCRIPTION
**Pull request by dangleptr**
Some interfaces have been changed in the latest version of folly and fbthrift. 
Let's modify them to avoid broken under latest dependencies.  

Subtask of #1160

### CODE REVIEW

File: src/kvstore/raftex/RaftPart.cpp [hunk 1]
```diff
@@ -775,7 +775,7 @@ void RaftPart::replicateLogs(folly::EventBase* eb,
             return resp.get_error_code() == cpp2::ErrorCode::SUCCEEDED
                     && !hosts[index]->isLearner();
         })
-        .then(executor_.get(), [self = shared_from_this(),
+        .via(executor_.get()).then([self = shared_from_this(),
```
Comment (darionyaphet) [inline src/kvstore/raftex/RaftPart.cpp]: Should using `thenValue` to replace `then` ?

Comment (dangleptr) [reply src/kvstore/raftex/RaftPart.cpp]: Because we need to handle the exception in folly::Try,  so we should use "then" in this place.

**Comment by zhangguoqing** [comment]
src/meta/test/TestUtils.h +131
allHosts.emplace_back(apache::thrift::FragileConstructor::FRAGILE, i, i);

maybe some others in ```grep FRAGILE -rn ./src/```

**Comment by dangleptr** [comment]
Jenkins go

**Pull request merged**